libafl/src/mutators/tuneable.rs

```diff
@@ -233,6 +247,36 @@ where
     }
 }
 
+// Returns the cumulative distribution function for a discrete distribution.
+fn get_cdf(probabilities: Vec<f32>) -> Result<Vec<f32>, Error> {
```

Comment:
I am not sure if this is OK here or if it should be in bolts.

Reply:
Not sure if it should move the `Vec` in and out and not just take a `&mut`? Both are somewhat correct, but this feels like it would alloc a new one(?)

Also, rust commonly doesn't use a `get_` prefix. Maybe `calculate_`?

Reply:
Feel free to move it to bolts and make it `pub` or leave it here as local fn, up to you :)

Reply:
> Not sure if it should move the Vec in and out and not just take a &mut? Both are somewhat correct, but this feels like it would alloc a new one(?)

I followed the same convention that we were using for the cumulative sum (move), but I don't really feel strongly about it. We were moving it before, I just kept the movement (?). Happy to change it.

I renamed the function to `calculate_cumulative_distribution`.

RE: Moving it to bolts, I am not sure someone else will use it, so for now it's fine here.